frame/support/src/traits.rs

```diff
@@ -1518,6 +1518,38 @@ pub trait OnFinalize<BlockNumber> {
 	fn on_finalize(_n: BlockNumber) {}
 }
 
+/// The block's on idle trait.
+///
+/// Implementing this lets you express what should happen for your pallet
+/// before on finalization, on_finalize hook, and based on if a remaining weight threshold is met.
```

Comment:
```suggestion
/// Implementing this lets you express what should happen for your pallet before
/// block finalization (see `on_finalize` hook) in case any remaining weight is left.
```